### PR TITLE
Migrate to JDA

### PR DESCRIPTION
**Pull request by JRoy**
JDA is actually maintained and has readable code and sane method names.

### CODE REVIEW

**Review by Bastian — CHANGES_REQUESTED** [review]
To be honest, I was hoping that you left more opportunities for me to shit-talk but over-all the PR's quality is quite good. 👍 

However, obviously I would advise against switching to JDA because Javacord is just the cooler library 😎 😉 .

This PR contains some smaller improvements that are independent from the library (e.g., using the `StringBuilder` where appropiate, making constant fields `final`, and many other small stuff). I would suggest to make a new PR that only has these changes without switching to JDA.


And to not keep your PR description uncommented:

> JDA is actually maintained [...]

So is Javacord. To be fair, JDA has some advantages by having a larger community.

> [...] and has readable code [...]

Look at the commit diff. Both libraries are actually pretty similar regarding the overall code.
The parts that are more different are usually more readable in Javacord because it is more concise (built-in message cache, sending messages directly to a user without opening a private channel first, directly adding images to embeds without having to upload them as an attachment, being able to attach listeners to objects, ...).

> [...] and sane method names.

Good joke. Just one example:

`embed.addField("title", "description", true)` vs `embed.addInlineField("title", "description")`.

Now tell me which one is more readable? As someone that has never worked with JDA or Javacord, you have no clue what `true` is refering to. Method names should be understandable without having to read the JavaDoc first. There are many similar examples in JDA andJavacord, e.g. `JDA#getUsersByName("Joe", true)` vs. `DiscordApi#getUsersByNameIgnoreCase("Joe")`. Just imagine if Java had a boolean flag for `String#equals()` to set case sensitivity 😉

File: src/main/java/org/moss/discord/util/EmbedPaged.java [hunk 1]
```diff
@@ -73,35 +74,24 @@ public EmbedPaged setPagedButtons(boolean bool) {
...
+                        message.addReaction(NEXT_PAGE_EMOJI).queue();
                     }
-                } else {
-                    message.addReaction(PREV_PAGE_EMOJI);
-                    message.addReaction(NEXT_PAGE_EMOJI);
+                    message.getJDA().addEventListener(instance);
+                    future.complete(message);
                 }
-                message.addReactionAddListener(this::onReactionClick);
-                message.addReactionRemoveListener(this::onReactionClick);
```
[the remark this answers is not in the document]
Comment (JRoy) [reply src/main/java/org/moss/discord/util/EmbedPaged.java:90]: Yes the option is nice but I also like ListenerAdapter :P

**Comment by JRoy** [comment]
Closing in favor of just PR'ing code improvements later

File: src/main/java/org/moss/discord/util/EmbedPaged.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Bastian) [reply src/main/java/org/moss/discord/util/EmbedPaged.java:90]: You like [this](https://github.com/DV8FromTheWorld/JDA/blob/master/src/main/java/net/dv8tion/jda/api/hooks/ListenerAdapter.java)?

**Pull request closed without merging**